docs/DeveloperGuide.md

```diff
@@ -4,10 +4,220 @@
+### [Proposed] Appointment Task-Employee Allocation feature
 
+### Proposed Implementation
+
+The proposed task allocation mechanism is facilitated by `AppointmentList` , `TaskList` and `EmployeeList`. It implements the following operations:
+
+- `AppointmentList#addTaskToAppointment()` —  Allocates a task to an appointment when task is created
+- `TaskList#createTask()` —  Creates task to be performed for an existing appointment, the task will be allocated to an employee during creation.
+- `TaskList#reallocateTask()` — Reallocates the employee in charge of a particular task.
+- `EmployeeList#viewIndividualTask()` — User can view the Tasks that a particular employee is in charge of
+
+Given below is an example usage scenario and how the Task-Employee mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The user executes `task add i/1 d/Wash Equipment t/12pm` command which calls `TaskList#createTask()` and creates a task, to wash equipment at 12pm. The `task add` command also calls `AppointmentList#addTaskToAppointment()` to allocate the task to Appointment 1. The user is prompted with a list of employees to allocate this task, and the user executes `1` to allocate the task to ‘Sally’.
+
+Step 2. The user executes `task view` command to view all the current tasks for the clinic. The command calls `TaskList#listTasks()` to display all existing tasks and the employee allocation.
+
+Step 3. The user wishes to reallocate task 1 to another employee and executes `task reallocate i/1` . The `task reallocate` command also calls `TaskList#reallocateTask()`, and the user is prompted with a list of employees to reallocate this task, and the user executes `2` to reallocate the task from Employee 1 - ‘Sally’ to Employee 2 - ‘John’.
+
+> The following sequence diagram shows how the task allocation/reallocation operation works:
+
+Step 4. The user then executes the command `employee task i/2` to view the tasks of employee 2, which is ‘John’ in this case. The `employee task` command calls `EmployeeList#viewIndividualTask()` . Due to the reallocation done in step 3, the task 1 is moved from Employee 2 - ‘John’ instead of Employee 1 - ‘Sally’, and task 1 will not show up in Employee 1’s individual tasks.
+
+### Design considerations:
+
+**Aspect: How reallocateTask executes:**
+
+- Alternative 1 (current choice):
+
+   Updates the Employee attribute of Task to another employee’s id.
+
+  - Pros: Less complex, `EmployeeList#viewIndividualTask()` will find from a single `TaskList` to retrieve that Employee’s list of tasks.
+  - Cons: If the list of tasks is too long, fetch speed may be slow.
+
+- Alternative 2:
+
+   Shift the task from one employee to another.
+
+  - Pros: Instant load speed when `EmployeeList#viewIndividualTask()` is called.
+  - Cons: Difficult to implement.
+  - Cons: Complex to manage multiple TaskLists belonging to different employees
+
+*{more aspects and alternatives to be added}*
```

Comment:
Great description of the pros and cons of the design, it helps the readers to understand why certain structure is chosen.